src/main/java/com/radixdlt/atommodel/system/SystemConstraintScrypt.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.atommodel.system;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.TypeToken;
+import com.radixdlt.atomos.ConstraintScrypt;
+import com.radixdlt.atomos.ParticleDefinition;
+import com.radixdlt.atomos.Result;
+import com.radixdlt.atomos.SysCalls;
+import com.radixdlt.constraintmachine.PermissionLevel;
+import com.radixdlt.constraintmachine.Spin;
+import com.radixdlt.constraintmachine.TransitionProcedure;
+import com.radixdlt.constraintmachine.TransitionToken;
+import com.radixdlt.constraintmachine.UsedCompute;
+import com.radixdlt.constraintmachine.VoidUsedData;
+import com.radixdlt.constraintmachine.WitnessValidator;
+import com.radixdlt.constraintmachine.WitnessValidator.WitnessValidatorResult;
+import java.util.Optional;
+
+/**
+ * Allows for the update of the epoch, timestamp and view state.
+ * Currently there is only a single system particle that should be in
+ * existence.
+ * TODO: use a non-radix-address path to store this system info
+ */
+public final class SystemConstraintScrypt implements ConstraintScrypt {
+	private static Result staticCheck(SystemParticle systemParticle) {
+		if (systemParticle.getEpoch() < 0) {
+			return Result.error("Epoch is less than 0");
+		}
+
+		if (systemParticle.getTimestamp() < 0) {
+			return Result.error("Timestamp is less than 0");
+		}
+
+		if (systemParticle.getView() < 0) {
+			return Result.error("View is less than 0");
+		}
+
+		return  Result.success();
+	}
+
+	@Override
+	public void main(SysCalls os) {
+		os.registerParticle(SystemParticle.class, ParticleDefinition.<SystemParticle>builder()
+			.addressMapper(p -> ImmutableSet.of())
+			.staticValidation(SystemConstraintScrypt::staticCheck)
+			.virtualizeSpin(p -> p.getView() == 0 && p.getEpoch() == 0 && p.getTimestamp() == 0 ? Spin.UP : null)
+			.build()
+		);
+
+		os.createTransition(
+			new TransitionToken<>(SystemParticle.class, TypeToken.of(VoidUsedData.class), SystemParticle.class, TypeToken.of(VoidUsedData.class)),
+			new TransitionProcedure<SystemParticle, VoidUsedData, SystemParticle, VoidUsedData>() {
+				@Override
+				public PermissionLevel requiredPermissionLevel() {
+					return PermissionLevel.SUPER_USER;
+				}
+
+				@Override
+				public Result precondition(SystemParticle inputParticle, VoidUsedData inputUsed, SystemParticle outputParticle,
+					VoidUsedData outputUsed) {
+
+					if (inputParticle.getEpoch() == outputParticle.getEpoch()) {
+						if (inputParticle.getView() >= outputParticle.getView()) {
+							return Result.error("Next view must be greater than previous.");
+						}
+
+						return Result.success();
+					}
```

Comment:
I think this would be less confusing and less code as an `if (...) { ... } else if (...) { ... }`, like this:
```
	if (inputParticle.getEpoch() == outputParticle.getEpoch()) {
		if (inputParticle.getView() >= outputParticle.getView()) {
			return Result.error("Next view must be greater than previous.");
		}
	} else if (inputParticle.getEpoch() + 1 != outputParticle.getEpoch()) {
		return Result.error("Bad next epoch");
	}
	return Result.success();
```